### PR TITLE
Add API routes for job group parents

### PR DESCRIPTION
**Pull request by Martchus**
* Works in the same way as the API for
  regular groups and shares the same code
* When deleting a parent group, all groups
  inside remain parentless
* Still TODO
  * List all job groups inside a parent
  * List all jobs inside a group

### CODE REVIEW

File: lib/OpenQA/Schema/Result/JobGroups.pm [hunk 1]
```diff
@@ -99,7 +99,7 @@ around 'keep_results_in_days' => sub {
 
 around 'keep_important_results_in_days' => sub {
     my ($orig, $self) = @_;
-    return $self->get_column('keep_important_results_in_days') // ($self->parent ? $self->parent->default_keep_results_in_days : OpenQA::Schema::JobGroupDefaults::KEEP_IMPORTANT_RESULTS_IN_DAYS);
+    return $self->get_column('keep_important_results_in_days') // ($self->parent ? $self->parent->default_keep_important_results_in_days : OpenQA::Schema::JobGroupDefaults::KEEP_IMPORTANT_RESULTS_IN_DAYS);
```
Comment (okurz) [inline lib/OpenQA/Schema/Result/JobGroups.pm]: Seems unrelated

File: lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm [hunk 2]
```diff
@@ -17,89 +17,110 @@ package OpenQA::WebAPI::Controller::API::V1::JobGroup;
 use Mojo::Base 'Mojolicious::Controller';
 use OpenQA::Schema::Result::JobGroups;
 
-sub list {
+# Helper methods
+
+sub is_parent {
+    my ($self) = @_;
+    return $self->req->url->path =~ qr/.*\/parent_groups/ ? 1 : 0;
```
Comment (coolo) [inline lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm]: why are you casting the match to a number? returning of the return of the =~ would do.

File: lib/OpenQA/Schema/Result/JobGroups.pm (hunk 1 above)
Comment (coolo) [inline lib/OpenQA/Schema/Result/JobGroups.pm]: how is it unrelated? you notice such bugs when you write tests for the functionality.

File: t/api/10-jobgroups.t [hunk 3]
```diff
@@ -125,15 +127,58 @@ subtest 'create' => sub() {
...
+            keep_important_logs_in_days => 45,
+            parent_id                   => $new_id,
+        });
+
+    my $get = $t->get_ok('/api/v1/job_groups/1001')->status_is(200);
+    is_deeply(
+        $get->tx->res->json,
+        [
+            {
+                'name'                           => 'opensuse',
```
Comment (coolo) [inline t/api/10-jobgroups.t]: as you have it 8 lines further above without the quotes in hashes, please unify this. Even if we don't enforce perlcritic for t/*
